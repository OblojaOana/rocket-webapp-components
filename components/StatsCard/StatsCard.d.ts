import { CardProps } from '@material-ui/core/Card';
import { CardContentProps } from '@material-ui/core/CardContent';
import { CardHeaderProps } from '@material-ui/core/CardHeader';
import { CardActionsProps } from '@material-ui/core/CardActions';

export interface Props {

    /**
     * The icon to be displayed. This property is required
     */
    Icon: unknown;

    /**
     * @default "blue"
    * The color of the icon
     */
    iconColor?: "orange" |
    "green" |
    "red" |
    "blue" |
    "purple"

    /**
     * The tile of the card
     */
    title?: React.ReactNode;

    /**
     * Description of the card content
     */
    description?: React.ReactNode;

    /**
     * Footer for description
     */
    Small?: React.ReactNode;

    /**
     * Icon for content
     */
    StatIcon?: unknown;

    /**
     * @default "gray"
     * The color of the status icon
     */
    statIconColor?: "warning" |
                    "primary" |
                    "danger" |
                    "success" |
                    "info" |
                    "rose" |
                    "gray"

    /**
     * Link for content
     */
    statLink?: Object;

    /**
     * Text for content
     */
    statText?: React.ReactNode;

    /**
    * Demos:
    * 
    * - https://bit.dev/totalsoft_oss/react-mui/stats-card
    * 
    *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/develop/components/StatsCard/StatsCard.js
    */

}

export type StatsCardProps = Props | CardProps | CardContentProps | CardHeaderProps | CardActionsProps

export default function StatsCard(props: StatsCardProps): JSX.Element;