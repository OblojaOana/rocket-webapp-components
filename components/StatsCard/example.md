##### Copy this snapshot in the overview panel

```js
import React from "react";
import StatsCard from "@bit/totalsoft_oss.react-mui.stats-card";
import Work from "@material-ui/icons/Work";
import Update from "@material-ui/icons/Update";

export default (
	<StatsCard
		Icon={Update}
		iconColor="green"
		title="Shrimp and Chorizo Paella"
		statText={"Status text"}
		StatIcon={Work}
		description={"This impressive paella is a perfect party dish and a fun meal to cook together with your guests. Add 1 cup of frozen peas along with the mussels, if you like."}
	/>
);
```


##### Copy this snapshot in the overview panel

```js
import React from "react";
import StatsCard from "@bit/totalsoft_oss.react-mui.stats-card";
import Work from "@material-ui/icons/Work";
import Update from "@material-ui/icons/Update";

export default (
	<StatsCard
		Icon={Update}
		iconColor="red"
		title="Shrimp and Chorizo Paella"
		statText={"Status text"}
		StatIcon={Work}
		description={"This impressive paella is a perfect party dish and a fun meal to cook together with your guests. Add 1 cup of frozen peas along with the mussels, if you like."}
	/>
);
```