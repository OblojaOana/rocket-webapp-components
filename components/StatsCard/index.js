import StatsCard from "../../core/StatsCard";
export default StatsCard;