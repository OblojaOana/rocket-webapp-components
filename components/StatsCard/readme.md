##### Copy this snapshot in the overview panel

```js
import React from "react";
import StatsCard from "@bit/totalsoft_oss.react-mui.stats-card";
import Work from "@material-ui/icons/Work";
import Update from "@material-ui/icons/Update";

export default (
  <StatsCard
    Icon={Update}
    iconColor="green"
    title={"Title"}
    StatIcon={Work}
    description={"Description"}
    statText={"State text"}
  />
);
```
