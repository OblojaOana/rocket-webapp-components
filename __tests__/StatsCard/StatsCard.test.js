import React from "react";
import renderer from "react-test-renderer";
import StatsCard from "../../components/StatsCard/StatsCard";
import Work from "@material-ui/icons/Work";
import Update from "@material-ui/icons/Update";

describe("StatsCard", () => {
  it("Snapshot", () => {
    const component = renderer.create(
      <StatsCard
        Icon={Update}
        iconColor="green"
        title={"Title"}
        StatIcon={Work}
        description={"Description"}
        statText={"State text"}
      />
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
