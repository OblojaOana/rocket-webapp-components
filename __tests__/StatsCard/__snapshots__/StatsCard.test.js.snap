// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard Snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root makeStyles-cardHeader-2 makeStyles-greenCardHeader-4"
  >
    <div
      className="MuiCardHeader-avatar makeStyles-cardAvatar-10"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-cardIcon-9"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3v7.12zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8h1.5z"
        />
      </svg>
    </div>
    <div
      className="MuiCardHeader-content"
    />
  </div>
  <div
    className="MuiCardContent-root makeStyles-cardContent-8"
  >
    <p
      className="MuiTypography-root makeStyles-cardCategory-11"
    >
      Title
    </p>
    <h2
      className="MuiTypography-root makeStyles-cardTitle-12 MuiTypography-h5"
    >
      Description
    </h2>
  </div>
  <div
    className="MuiCardActions-root makeStyles-cardActions-14 MuiCardActions-spacing"
  >
    <div
      className="makeStyles-cardStats-15"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-cardStatsIcon-16 makeStyles-grayCardStatsIcon-23"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
        />
      </svg>
       
    </div>
  </div>
</div>
`;
